product_supplierinfo_for_customer_invoice/models/account_invoice.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2013-2017 Agile Business Group sagl
+#     (<http://www.agilebg.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class AccountInvoiceLine(models.Model):
+    _inherit = 'account.invoice.line'
+
+    @api.multi
+    def _get_product_customer_code(self):
+        product_supplierinfo_obj = self.env['product.supplierinfo']
+        for line in self.filtered(lambda il: il.product_id.supplier_ids):
+            product = line.product_id
+            code_id = product_supplierinfo_obj.search([
```

Comment:
Does not look good to me. I created a customer invoice and the customer code in the invoice was totally incorrect.

It should filter by the customer, and also the product.supplierinfo must be type customer